docs/src/pages/components/tree-view/CustomizedTreeView.js

```diff
@@ -54,40 +70,44 @@ TransitionComponent.propTypes = {
   in: PropTypes.bool,
 };
 
-const StyledTreeItem = withStyles(theme => ({
```

Comment:
I think that we should use withStyles here. It's the approach we document for customizing components. They might be a case or two where we don't do it, that was because I couldn't get TypeScript to accept it.

Reply:
> It's the approach we document for customizing components

Perhaps we should update the documentation then? The hook method is much cleaner.

> we shouldn't share the same style sheet

I did it separately originally, and thought you'd say to combine them! 😆

What's the main concern?

Reply:
> What's the main concern?

I think that people should be able to copy & paste the customized tree item and to move it into its own file without having to worry about the styles of the overall tree view.

> Perhaps we should update the documentation then? The hook method is much cleaner.

I believe that we document makeStyles for page layout styles (but can often be replaced with the Box) and for new components creation. I believe that we also document withStyles for customizing existing components. Migrating to styled-components, we will have to remove all the hook styles.

Reply:
> move it into its own file without having to worry about the styles

Okay, that's what I guessed.

Reply:
>  Migrating to styled-components, we will have to remove all the hook styles.

Sounds like a migration nightmare for users.

Reply:
I think that the API should still work, my comment was targeted at what we use in the demos.